src/Compilers/CSharp/Portable/FlowAnalysis/DefiniteAssignment.cs

```diff
@@ -1685,6 +1685,13 @@ protected override void EnterParameter(ParameterSymbol parameter)
                 if (slot > 0) SetSlotState(slot, true);
                 NoteWrite(parameter, value: null, read: true);
             }
+
+            if (parameter.ContainingSymbol is LocalFunctionSymbol or LambdaSymbol &&
+                    parameter.ExplicitDefaultConstantValue?.IsBad != true &&
```

Comment:
You can actually simplify this further:
```suggestion
            if (parameter is { ContainingSymbol: LocalFunctionSymbol or LambdaSymbol, ExplicitDefaultConstantValue.IsBad: not true } &&
```

Reply:
Actually, thinking about it further, we probably want to always look at the value, even if it's bad, as that will remove spurious diagnostics. Please add a test for a non-const local that's only referenced in a lambda/local function parameter.

Reply:
> we probably want to always look at the value, even if it's bad, 

Second that